lib/auth/auth_with_roles.go

```diff
@@ -3251,6 +3251,12 @@ func (a *ServerWithRoles) generateUserCerts(ctx context.Context, req proto.UserC
 		connectionDiagnosticID: req.ConnectionDiagnosticID,
 		attestationStatement:   keys.AttestationStatementFromProto(req.AttestationStatement),
 		botName:                getBotName(user),
+
+		// Always pass through a bot instance ID if available. Note that this
+		// method is only used for bot identity renewals and is not responsible
+		// for issuing new instance IDs; see `generateInitialBotCerts()`
+		// TODO: need to update bot instance with new authentication + generation counter
```

Comment:
We'll need to construct a new authentication record here, and I think it creates a couple weird cases I'm not certain of:
* should we add an auth record just for renewals, or also for impersonated certs? (I assume the former)
* what is the join token value for this path? Should it just be `""`?
* what will metadata be? empty?

Re-authenticated delegated joins should all go through the `generateInitialBotCerts()` route rather than here, so this _should_ just be `token` method renewals, assuming we don't include impersonated certs.

Reply:
> should we add an auth record just for renewals, or also for impersonated certs? (I assume the former)

Definitely just for renewals and not for impersonated certs. It'll be much noisier if we do impersonated certs.

> what is the join token value for this path? Should it just be ""?

Yeah - I'd think so. 

> what will metadata be? empty?

Yeah - empty - at least in terms of identity metadata. We could include information like the generation counter etc but I'm not convinced it really adds anything compared to the rich metadata we get from somewhere like GitHub.

Reply:
Works for me. We do have the generation counter in the auth status proto already, so I can't think of anything else we'd want to add.